pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -555,10 +600,24 @@ func (m *kubeGenericRuntimeManager) computePodActions(pod *v1.Pod, podStatus *ku
 		return changes
 	}
 
+	//check sidecars status
+	sidecarStatus := status.GetSidecarsStatus(pod)
+
 	// Number of running containers to keep.
 	keepCount := 0
 	// check the status of containers.
 	for idx, container := range pod.Spec.Containers {
+
+		if utilfeature.DefaultFeatureGate.Enabled(features.SidecarLifecycle) {
```

Comment:
Can you add a comment here along the lines of... "in practice, this check guarantees that sidecar containers will be started before normal containers"?

Reply:
yeah will do